src/game/frontend/submenus/World/ObjectSpawner.cpp

```diff
@@ -0,0 +1,154 @@
+#include "game/rdr/data/ObjModels.hpp"
+
+#include "ObjectSpawner.hpp"
+
+namespace YimMenu::Submenus
+{
+	static bool IsObjectModelInList(const std::string& model)
+	{
+		for (const auto& objModel : YimMenu::Data::g_ObjModels)
+		{
+			if (objModel.model == model)
+				return true;
+		}
+		return false;
+	}
+
+	static int ObjectSpawnerInputCallback(ImGuiInputTextCallbackData* data)
+	{
+		if (data->EventFlag == ImGuiInputTextFlags_CallbackCompletion)
+		{
+			std::string newText;
+			std::string inputLower = data->Buf;
+			std::transform(inputLower.begin(), inputLower.end(), inputLower.begin(), ::tolower);
+
+			for (const auto& objModel : YimMenu::Data::g_ObjModels)
+			{
+				std::string modelLower = objModel.model;
+				std::transform(modelLower.begin(), modelLower.end(), modelLower.begin(), ::tolower);
+				if (modelLower.find(inputLower) != std::string::npos)
+				{
+					newText = objModel.model;
+					break;
+				}
+			}
+
+			if (!newText.empty())
+			{
+				data->DeleteChars(0, data->BufTextLen);
+				data->InsertChars(0, newText.c_str());
+			}
+
+			return 1;
+		}
+		return 0;
+	}
+
+	void RenderObjectSpawnerMenu()
+	{
+		static std::string objectModelBuffer;
+		static float positionOffsetX = 5.0f;
+		static float positionOffsetY = 5.0f;
+		static float positionOffsetZ = 0.0f;
+		static bool isDynamic        = false;
+		static int modelHash         = 0;
+
+		char buffer[256];
+		strncpy(buffer, objectModelBuffer.c_str(), sizeof(buffer));
+
+		ImGui::InputTextWithHint("##objectmodel", "Object Model", buffer, sizeof(buffer), ImGuiInputTextFlags_CallbackCompletion, ObjectSpawnerInputCallback);
+
+		objectModelBuffer = buffer;
+
+		if (ImGui::IsItemHovered())
+			ImGui::SetTooltip("Press Tab to auto-fill");
+
+		if (!objectModelBuffer.empty() && !IsObjectModelInList(objectModelBuffer))
+		{
+			ImGui::BeginListBox("##objectmodels", ImVec2(250, 100));
+
+			std::string bufferLower = objectModelBuffer;
+			std::transform(bufferLower.begin(), bufferLower.end(), bufferLower.begin(), ::tolower);
+			for (const auto& objModel : YimMenu::Data::g_ObjModels)
+			{
+				std::string objModelLower = objModel.model;
+				std::transform(objModelLower.begin(), objModelLower.end(), objModelLower.begin(), ::tolower);
+				if (objModelLower.find(bufferLower) != std::string::npos && ImGui::Selectable(objModel.model.c_str()))
+				{
+					objectModelBuffer = objModel.model;
+					modelHash         = Joaat(objectModelBuffer.c_str());
+					STREAMING::REQUEST_MODEL(modelHash, false);
+					while (!STREAMING::HAS_MODEL_LOADED(modelHash))
+					{
+						return;
+					}
+				}
+			}
+
+			ImGui::EndListBox();
+		}
+
+		ImGui::SliderFloat("Offset X", &positionOffsetX, -25.0f, 25.0f);
+		ImGui::SliderFloat("Offset Y", &positionOffsetY, -25.0f, 25.0f);
+		if (isDynamic == false)
+		{
+			ImGui::SliderFloat("Offset Z", &positionOffsetZ, -10.0f, 10.0f);
+		}
+		ImGui::Checkbox("Dynamic", &isDynamic);
+		if (ImGui::Button("Spawn Object"))
```

Comment:
Use the function for spawning objects located in `util/SpawnObject.cpp`

Reply:
Using this forces the object to be placed on the ground properly, which renders the Z coordinate slider totally useless when that native is active.  I also cannot use that function to set my objectHandle (cannot convert void SpawnObject), which is needed to make things like the gold/cash boxes able to be picked up and obtained.